src/extension.ts

```diff
@@ -481,6 +481,10 @@ async function activate(context: vscode.ExtensionContext) {
     );
   }
 
+  if (await connector.shouldAutoConnect()) {
```

Comment:
There might be an issue here. `shouldAutoConnect` returns a promise that resolves to `nReplPortFileExists()` if the `autoConnectRepl` config is true. `nReplPortFileExists` returns a promise that resolves to an array of true and false values, so this `if` can evaluate to something like `if ([false, false, false]) ...` if no port file exists, which would be a true statement. Then the connect command would be run even though there's no port file.

![image](https://user-images.githubusercontent.com/12722744/224442415-4bc00332-aa9e-4d68-b8d2-32c8bb6a6bc2.png)

Reply:
I'll look again at this.

Reply:
I think that both these should be taken care of now.

Reply:
It doesn't resolve to an array any longer.

Reply:
Oh, gotcha!